internal/pkg/docker/dockerfile/dockerfile_test.go

```diff
@@ -196,3 +192,97 @@ EXPOSE 8080/tcp 5000`),
 		})
 	}
 }
+
+func TestParseHealthCheckDockerfile(t *testing.T) {
+	testCases := map[string]struct {
+		dockerfile   string
+		err          error
+		wantedConfig Dockerfile
+	}{
+		"correctly parses healthcheck with default values": {
+			dockerfile: `HEALTHCHECK CMD curl -f http://localhost/ || exit 1`,
+			err:        nil,
+			wantedConfig: Dockerfile{
+				HealthCheck: healthCheck{
+					interval:    30,
+					timeout:     30,
+					startPeriod: 0,
+					retries:     3,
+					command:     "CMD curl -f http://localhost/ || exit 1",
+				},
+				ExposedPorts: []portConfig{},
+			},
+		},
+		"correctly parses healthcheck with user's values": {
+			dockerfile: `HEALTHCHECK --interval=5m --timeout=3s --start-period=2s --retries=3 \
+			CMD curl -f http://localhost/ || exit 1`,
+			err: nil,
+			wantedConfig: Dockerfile{
+				HealthCheck: healthCheck{
+					interval:    300,
+					timeout:     3,
+					startPeriod: 2,
+					retries:     3,
+					command:     "CMD curl -f http://localhost/ || exit 1",
+				},
+				ExposedPorts: []portConfig{},
+			},
+		},
+		"correctly parses healthcheck with NONE": {
+			dockerfile: `HEALTHCHECK NONE`,
+			err:        nil,
+			wantedConfig: Dockerfile{
+				HealthCheck: healthCheck{
+					interval:    0,
+					timeout:     0,
+					startPeriod: 0,
+					retries:     0,
+					command:     "",
+				},
+				ExposedPorts: []portConfig{},
+			},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			got, _ := parseFromReader(tc.dockerfile)
+			require.Equal(t, tc.wantedConfig, got)
+		})
+	}
+}
+
+func TestParseHealthCheckErrorDockerfile(t *testing.T) {
```

Comment:
it seems like this test func calls `parseFromReader` as well. Why not combine this one with `TestParseHealthCheckDockerfile` and `TestParseExposeDockerfile` like we did in other pkgs?

Reply:
Really nice tests, the test cases are really good! 
+1 to what PH said, but also we should only write tests against the public interface of our packages. So instead of calling `parseFromReader`, it'd be neat to call `df.GetHealthCheck()` instead.

Reply:
I was going to write that method for parsing the healthcheck options to the manifest but I can do that this PR and use it for testing.